src/sentry/management/commands/makemigrations.py

```diff
@@ -15,14 +16,34 @@
 """
 
 
+# We check that the latest migration is the one stored in the lockfile
+def validate(migrations_file, latest_migration_by_app):
+    infile = {}
+    with open(migrations_file, encoding="utf-8") as file:
+        for line in file:
+            try:
+                app_label, name = line.split(": ")
+                infile[app_label] = name.strip()
+            except ValueError:
+                pass
+
+    for app_label, name in sorted(latest_migration_by_app.items()):
+        if infile[app_label] != name:
+            print(  # noqa: B314
+                f"ERROR: The latest migration does not match the contents of the {app_label}: {name} vs {infile[app_label]}"
+            )
+            # makemigrations.Command --check exits with 1 if a migration needs to be generated
+            sys.exit(2)
```

Comment:
Internally when we call --check it will exit with 1 if migrations needed to be generated. This is to differentiate from the case Billy handles in the CI.